src/textual/reactive.py

```diff
@@ -50,6 +53,108 @@ class TooManyComputesError(ReactiveError):
     """Raised when an attribute has public and private compute methods."""
 
 
+class WatchDecorator(Generic[WatchMethodType]):
+    """Watch decorator.
+
+    Decorate a method to make it a watcher.
+
+    """
+
+    def __init__(self, reactive: Reactive | None = None) -> None:
+        self._reactive = reactive
+
+    @overload
+    def __call__(self, *, init: bool = True) -> WatchDecorator[WatchMethodType]: ...
+
+    @overload
+    def __call__(self, method: WatchMethodType) -> WatchMethodType: ...
+
+    def __call__(
+        self, method: WatchMethodType | None = None, *, init: bool = True
+    ) -> WatchMethodType | WatchDecorator[WatchMethodType]:
+        _rich_traceback_omit = True
+        if method is None:
+            return self
+        assert self._reactive is not None
+        assert hasattr(method, "__name__")
+        if not method.__name__.startswith("watch_"):
+            if self._reactive._watch_method is not None:
+                raise RuntimeError(
```

Comment:
When this is raised, we don't get the pretty-printed Textual tracebacks.
I'm assuming you're aware of that?

![Screenshot 2024-02-27 at 16 20 00](https://github.com/Textualize/textual/assets/5621605/9be26ddc-c303-49d4-8cc2-f2524de7321b)

Similar comments apply to the compute and validate decorators.

Reply:
Yes, because it occurs at import time, before the traceback hooks. I don't want to set up the pretty tracebacks at import time, because that would force it on users who just import `textual`.

Reply:
Ok, thanks for explaining.